odoo/addons/base/models/ir_ui_view.py

```diff
@@ -1150,9 +1150,27 @@ def _validate_tag_search(self, node, name_manager, node_info):
                 _('Search tag requires at least one field element'), raise_exception=False)
 
     def _validate_tag_searchpanel(self, node, name_manager, node_info):
+        Model = node_info['attr_model']
+        fields = Model._fields
         for child in node.iterchildren(tag=etree.Element):
-            if child.get('domain') and child.get('select') != 'multi':
-                msg = _('Searchpanel item with select multi cannot have a domain.')
+            field_name = child.get('name')
+            field_type = fields[field_name].type
+            select = child.get('select', 'one')
+            if select == 'one' and field_type not in ['many2one', 'selection']:
+                msg = _('A search panel field with select="one" (default) must have many2one or selection type.')
+                self.handle_view_error(msg)
+            if select == 'multi' and field_type not in ['many2one', 'many2many', 'selection']:
+                if not (Model._name == 'documents.document' and field_name == 'res_model'):
```

Comment:
This is the kind of problem we can run into. We do not let overrides manage other field types. I think this is why we first put restrictions in the generic code of models.py.